rust/datafusion/src/physical_plan/parquet.rs

```diff
@@ -209,6 +251,479 @@ impl ParquetExec {
+                self.column_name.to_string(),
+                StatisticsType::Max,
+                max_field,
+            ));
+        }
+        max_column_name
+    }
+
+    fn build(&self, statistics_expr: &Expr) -> Result<Arc<dyn PhysicalExpr>> {
+        let execution_context_state = ExecutionContextState {
+            datasources: HashMap::new(),
+            scalar_functions: HashMap::new(),
+            var_provider: HashMap::new(),
+            aggregate_functions: HashMap::new(),
+            config: ExecutionConfig::new(),
+        };
+        let schema = Schema::new(self.statistics_fields.clone());
+        DefaultPhysicalPlanner::default().create_physical_expr(
+            statistics_expr,
+            &schema,
+            &execution_context_state,
+        )
+    }
+}
+
+/// Translate logical filter expression into parquet statistics physical filter expression
+fn build_predicate_expression(
+    expr: &Expr,
+    parquet_schema: &Schema,
+    stat_column_req: &mut Vec<(String, StatisticsType, Field)>,
+) -> Result<Arc<dyn PhysicalExpr>> {
+    // predicate expression can only be a binary expression
+    let (left, op, right) = match expr {
+        Expr::BinaryExpr { left, op, right } => (left, op, right),
+        _ => {
+            return Ok(expressions::lit(ScalarValue::Boolean(Some(true))));
+        }
+    };
+
+    if op == &Operator::And || op == &Operator::Or {
+        let left_expr =
+            build_predicate_expression(left, parquet_schema, stat_column_req)?;
+        let right_expr =
+            build_predicate_expression(right, parquet_schema, stat_column_req)?;
+        let stat_fields = stat_column_req
+            .iter()
+            .map(|(_, _, f)| f.clone())
+            .collect::<Vec<_>>();
+        let stat_schema = Schema::new(stat_fields);
+        let result = expressions::binary(left_expr, op.clone(), right_expr, &stat_schema);
+        return result;
+    }
+
+    let mut expr_builder = match PhysicalExpressionBuilder::try_new(
+        left,
+        right,
+        parquet_schema,
+        stat_column_req,
+    ) {
+        Ok(builder) => builder,
+        // allow partial failure in predicate expression generation
+        // this can still produce a useful predicate when multiple conditions are joined using AND
+        Err(_) => {
+            return Ok(expressions::lit(ScalarValue::Boolean(Some(true))));
+        }
+    };
+    let corrected_op = expr_builder.correct_operator(op);
+    let statistics_expr = match corrected_op {
+        Operator::Eq => {
+            let min_column_name = expr_builder.add_min_column();
+            let max_column_name = expr_builder.add_max_column();
+            // column = literal => column = (min, max) => min <= literal && literal <= max
+            // (column / 2) = 4 => (column_min / 2) <= 4 && 4 <= (column_max / 2)
+            expr_builder
+                .scalar_expr()
+                .gt_eq(rewrite_column_expr(
```

Comment:
stylistically, you might be able to hoist out the repeated calls to 

```
rewrite_column_expr(
                    expr_builder.column_expr(),
                    expr_builder.column_name(),
                    max_column_name.as_str()
```

and 

```
rewrite_column_expr(
                    expr_builder.column_expr(),
                    expr_builder.column_name(),
                    min_column_name.as_str()
```

 by evaluating them once before the `match` expression:

```
let min_col_expr = rewrite_column_expr(
                    expr_builder.column_expr(),
                    expr_builder.column_name(),
                    min_column_name.as_str());

let max_col_expr = rewrite_column_expr(
                    expr_builder.column_expr(),
                    expr_builder.column_name(),
                    max_column_name.as_str())
```

But they way you have it works well too

Reply:
@alamb I have just pushed a change which removes this repetition and makes that part of the code cleaner; not long to go now, some more tests to add for the execution of the row group predicate in the next couple of days and this work should be ready to merge

Reply:
Awesome -- thanks @yordan-pavlov  -- I am excited for this one. When it is ready I'll re-review the code and get it merged asap.

Thanks again for introducing this feature. 🎉